benchmarks/resource_aware_pool_stress_test.cpp

```diff
@@ -0,0 +1,171 @@
+#include <stdio.h>
+#include <math.h>
+#include <iostream>
+
+#include "camp/camp.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/strategy/QuickPool.hpp"
+#include "umpire/strategy/ResourceAwarePool.hpp"
+
+using namespace camp::resources;
+
+constexpr int ITER = 5;
+constexpr int NUM = 2048;
+const int NUM_PER_BLOCK = 256;
+const int NUM_BLOCKS = NUM/NUM_PER_BLOCK;
+
+using clock_value_t = long long;
+
+__device__ clock_value_t my_clock()
+{
+  return clock64();
+}
+
+__device__ void sleep(clock_value_t sleep_cycles)
+{
+  clock_value_t start = my_clock();
+  clock_value_t cycles_elapsed;
+  do {
+    cycles_elapsed = my_clock() - start;
+  } while (cycles_elapsed < sleep_cycles);
+}
+
+__global__ void do_sleep()
+{
+  // sleep - works still at 1000, so keeping it at 100k
+  sleep(100000000);
+}
+
+__global__ void touch_data(double* data)
+{
+  int id = blockIdx.x * blockDim.x + threadIdx.x;
+
+  if (id < NUM) {
+    data[id] = id;
+  }
+}
+
+__global__ void check_data(double* data)
+{
+  int id = blockIdx.x * blockDim.x + threadIdx.x;
+
+  //Then error check that data[id] still == id
+  if (id < NUM) {
+    if (data[id] != id)
+      data[id] = -1; 
+  }
+}
+
+__global__ void touch_data_again(double* data)
+{
+  int id = blockIdx.x * blockDim.x + threadIdx.x;
+
+  if (id < NUM) {
+    data[id] = 8.76543210;
+  }
+}
+
+void QuickPool_check(umpire::Allocator quick_pool)
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  bool error{false};
+
+  // Create hip streams
+  hipStream_t s1, s2;
+  hipStreamCreate(&s1); hipStreamCreate(&s2);
+
+  double* a = static_cast<double*>(quick_pool.allocate(NUM * sizeof(double)));
+
+  hipLaunchKernelGGL(touch_data, dim3(NUM_BLOCKS), dim3(NUM_PER_BLOCK), 0, s1, a);
+  hipLaunchKernelGGL(do_sleep, dim3(NUM_BLOCKS), dim3(NUM_PER_BLOCK), 0, s1);
+  hipLaunchKernelGGL(check_data, dim3(NUM_BLOCKS), dim3(NUM_PER_BLOCK), 0, s1, a);
+
+  quick_pool.deallocate(a);
+  a = static_cast<double*>(quick_pool.allocate(NUM * sizeof(double)));
+
+  hipLaunchKernelGGL(touch_data_again, dim3(NUM_BLOCKS), dim3(NUM_PER_BLOCK), 0, s2, a);
+
+  double* b = static_cast<double*>(quick_pool.allocate(NUM * sizeof(double)));
+  rm.copy(b, a);
```

Comment:
We're relying on this being implicitly in order with both s1 and s2, it would probably be better to just synchronize the device before doing the copy to ensure all work on s1 and s2 is finished before the copy.